src/A2A/Models/FileContent.cs

```diff
@@ -1,16 +1,58 @@
 using System.Text.Json;
 using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
 
 namespace A2A;
 
+/// <summary>
+/// Defines the set of FileContent kinds used as the 'kind' discriminator in serialized payloads.
+/// </summary>
+/// <remarks>
+/// Values are serialized as lowercase kebab-case strings via <see cref="KebabCaseLowerJsonStringEnumConverter{TEnum}"/>.
+/// </remarks>
+[JsonConverter(typeof(KebabCaseLowerJsonStringEnumConverter<FileContentKind>))]
+public enum FileContentKind
```

Comment:
What about nesting "kind" enums inside corresponding base data class?

Reply:
I don't understand. Please explain this question.